packages/storybook/src/ui/atoms/icons/icons.test.tsx

```diff
@@ -38,187 +38,221 @@ import {
+    [<ExternalLinkIcon size="small" data-testid="icon" className="further-maths" />, 'icon__external-link'],
+    [<FilterIcon size="small" data-testid="icon" className="further-maths" />, 'icon__filter'],
+    [<FolderPlusIcon size="small" data-testid="icon" className="further-maths" />, 'icon__folder__plus'],
+    [<GridLayoutIcon size="small" data-testid="icon" className="further-maths" />, 'icon__grid-layout'],
+    [<HamburgerIcon size="small" data-testid="icon" className="further-maths" />, 'icon__hamburger'],
+    [<InfoCircleIcon size="small" data-testid="icon" status="green" className="further-maths" />, 'icon__info-circle'],
+    [<KeyboardIcon size="small" data-testid="icon" className="further-maths" />, 'icon__keyboard'],
+    [<LayerGroupIcon size="small" data-testid="icon" className="further-maths" />, 'icon__layer__group'],
+    [<ListLayoutIcon size="small" data-testid="icon" className="further-maths" />, 'icon__list-layout'],
+    [<PauseIcon size="small" data-testid="icon" className="further-maths" />, 'icon__pause'],
+    [<PlayIcon size="small" data-testid="icon" className="further-maths" />, 'icon__play'],
+    [<PlusIcon size="small" data-testid="icon" className="further-maths" />, 'icon__plus'],
+    [<SpinnerIcon size="small" data-testid="icon" className="further-maths" />, 'icon__spinner'],
+    [<StopIcon size="small" data-testid="icon" className="further-maths" />, 'icon__stop'],
+    [<TableIcon size="small" data-testid="icon" status="default" className="further-maths" />, 'icon__table'],
+    [<ToggleOffIcon size="small" data-testid="icon" className="further-maths" />, 'icon__toggle--off'],
+    [<ToggleOnIcon size="small" data-testid="icon" className="further-maths" />, 'icon__toggle--on'],
+    [<TrashIcon size="small" data-testid="icon" className="further-maths" />, 'icon__trash'],
+    [<UserIcon size="small" data-testid="icon" className="further-maths" />, 'icon__user'],
+    [<DotCircleIcon size="small" data-testid="icon" status="default" className="further-maths" />, 'icon__dot__circle'],
+    [<SearchIcon size="small" data-testid="icon" className="further-maths" />, 'icon__search'],
+    [<CheckboxIcon size="small" data-testid="icon" className="further-maths" />, 'icon__checkbox'],
+    [<SquareIcon size="small" data-testid="icon" className="further-maths" />, 'icon__square'],
+    [
+      <ChevronDoubleIcon size="small" data-testid="icon" direction="up" className="further-maths" />,
+      'icon__chevron-double--up',
+    ],
+    [<ArrowIcon size="small" data-testid="icon" direction="up" className="further-maths" />, 'icon__arrow--up'],
   ])('Still has the right class name when given custom class names', (icon: JSX.Element, expectedClass: string) => {
     render(icon)
 
-    expect(screen.getByRole('img', { hidden: true })).toHaveClass(expectedClass)
-    expect(screen.getByRole('img', { hidden: true })).toHaveClass('further-maths')
```

Comment:
What's the reason for swapping these methods and adding the data test ids? 
I'd always prefer `getByRole` because it's implicitly asserting that a) the component has the correct role and b) that it holds the `hidden` property to remove it from the accessibility tree.
If the new emotion icons can't be found with this criteria then I'd prefer to re-factor them until they do

Reply:
If you check out the `Priority` section in the below link it explains why - basically there's an order of preference of things you look for, and data test id should only be used when you can't use anything else 
https://testing-library.com/docs/queries/about/

Reply:
emotion icons didn't have img role but i guess i can add that to the icon itself at atom level. going to fix that 👍